addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -50,6 +50,38 @@ def setUpClass(cls):
 
         cls.user_salemanager = new_test_user(cls.env, login='user_salemanager', groups='sales_team.group_sale_manager')
 
+        cls.promotion_code_10pc = cls.env['loyalty.program'].create({
+            'name': 'Code for 10% on orders',
+            'trigger': 'with_code',
+            'program_type': 'promotion',
+            'applies_on': 'current',
+            'rule_ids': [Command.create({
+                'mode': 'with_code',
+                'code': 'test_10pc',
+            })],
+            'reward_ids': [Command.create({
+                'reward_type': 'discount',
+                'discount_mode': 'percent',
+                'discount': 10,
+                'discount_applicability': 'order',
+                'required_points': 1,
+            })],
+        })
+
+    def create_programs(self, details):
+        """
+        Create loyalty programs based on the details given.
+        :param details: list of tuple ('name': str, 'program_type': 'gift_card' or 'ewallet')
+        """
+        LoyaltyProgram = self.env['loyalty.program']
+        programs = {}  # map: name -> program
+        for (name, program_type) in details:
+            program_id = LoyaltyProgram.create_from_template(program_type)['res_id']
+            program = LoyaltyProgram.browse(program_id)
+            program.write({'name': name})
+            programs[name] = program
+        return programs
```

Comment:
Sorry I've just re-used the method that exists in POS. I understand the first part about the single program but not so much that last part. I guess that you talk about using `cls.env['loyalty.program'].create()` but how will i be able to get the base setting configuration for the program type since it's not the same for all programs? 
If I simply do that: 
```python
`cls.env['loyalty.program'].create({
    'name': name;
    'program_type': program_type,
})`
```
it is enough?

Reply:
If the program's fields have no influence on the test, then I think you can indeed simply create a program with `name` and `program_type`. Alternatively, if you do need the base setup, you can do something like `LoyaltyProgram.create(LoyaltyProgram._get_template_values()[program_type])` (which does the same as what you have, but without creating an action).

If you do rely on some of the program's fields, then I would set them explicitly (so that it's clear where they come from, and so that your test doesn't break if someone changes them).